frontend-project/scripts/generate_interface.js

```diff
@@ -0,0 +1,41 @@
+const fs = require('fs');
+
+const main = async () => {
+    if (process.argv.length < 3) {
+        console.error(`${process.argv.slice(0, 2).join("")} [specification-file]`);
+        process.exit(-1);
+    }
+    const spec = JSON.parse(await fs.promises.readFile(process.argv[2]));
+    console.log('// This file is generated, do not manually edit this file');
+    console.log('// Update and use \'@/scripts/generate_interface.js\' instead');
+    console.log('');
+    for (const [name, schema] of Object.entries(spec.definitions)) {
+        console.log(`export interface ${name} {`);
+        for (const [pname, pvalue] of Object.entries(schema.properties)) {
+            if (pvalue.type == 'string') {
+                console.log(`  ${pname}: string;`);
+            } else if (pvalue.type == 'integer') {
+                console.log(`  ${pname}: number;`);
+            } else if (pvalue.type == 'boolean') {
+                console.log(`  ${pname}: boolean;`);
+            } else if (pvalue.type == 'array' && pvalue.items.type == 'integer') {
+                console.log(`  ${pname}: number[];`);
+            } else if (pvalue.type == 'array' && pvalue.items.type == 'string') {
+                console.log(`  ${pname}: string[];`);
+            } else if (pvalue.type == 'array' && pvalue.items['$ref']) {
+                const name = pvalue.items['$ref'].replace('#/definitions/', '');
+                console.log(`  ${pname}: ${name}[];`);
```

Comment:
w przypadku nested objects to chyba spowoduje błąd

Reply:
W jakim przypadku? Czy otrzymany wynik jest w jakiś sposób nieprawidłowy?

Reply:
@ad-m o ile się nie mylę to w przypadku gdy zagnieżdżony obiekt nie będzie tablicą
Nie sprawdzałem, czy tak to działa. Piszę chyba, bo nie wiem jak to wygląda w pliku openapi, wieczorem nad tym siądę

Reply:
Aaa, dobra, widzę w 30 linijce jest "any", co też nie jest najlepszym rozwiązaniem

Reply:
Any jest wtedy, gdy mamy tam praktycznie wszystko i tak jest stworzona specyfikacja OpenAPI, aby było tam potencjalnie wszystko.

Reply:
>  gdy zagnieżdżony obiekt nie będzie tablicą

Wyraźnie sprawdzam czy taki obiekt jest tablicą. Nie mamy zagnieżdzonych obiektów, które nie są tablica, więc nie zapisałem obsługi. Proponuje pierw przeanalizować wygenerowane wyjście.

Reply:
@ad-m any jest wyłączeniem typechecka w typescripcie, bardziej by coś w stylu unknown pasowało

co do tych obiektów to myślę, że mimo wszystko warto to wspierać, nawet jeśli nie teraz to w przyszłości

pisałem, że zmian możesz się spodziewać wieczorem ale mam parę pomysłów i z chęcią bym je przedyskutował